use-cases/vehicle-routing-time-windows/src/main/java/org/acme/vehiclerouting/domain/Location.java

```diff
@@ -0,0 +1,72 @@
+package org.acme.vehiclerouting.domain;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonFormat(shape = JsonFormat.Shape.ARRAY)
+public class Location {
+
+    private double latitude;
+    private double longitude;
+
+    @JsonIgnore
+    private Map<Location, Long> distanceMap;
+
+    @JsonCreator
+    public Location(@JsonProperty("latitude") double latitude, @JsonProperty("longitude") double longitude) {
+        this.latitude = latitude;
+        this.longitude = longitude;
+    }
+
+    public double getLatitude() {
+        return latitude;
+    }
+
+    public double getLongitude() {
+        return longitude;
+    }
+
+    public Map<Location, Long> getDistanceMap() {
+        return distanceMap;
+    }
+
+    /**
+     * Set the distance map. Distances are in meters.
+     *
+     * @param distanceMap a map containing distances from here to other locations
+     */
+    public void setDistanceMap(Map<Location, Long> distanceMap) {
+        this.distanceMap = distanceMap;
+    }
+
+    /**
+     * Distance to the given location in meters.
+     *
+     * @param location other location
+     * @return distance in meters
+     */
+    public long getDistanceTo(Location location) {
+        return distanceMap.get(location);
+    }
+
+    // ************************************************************************
+    // Complex methods
+    // ************************************************************************
+
+    /**
+     * The angle relative to the direction EAST.
+     *
+     * @param location never null
+     * @return in Cartesian coordinates
+     */
+    public double getAngle(Location location) {
+        // Euclidean distance (Pythagorean theorem) - not correct when the surface is a sphere
+        double latitudeDifference = location.latitude - latitude;
+        double longitudeDifference = location.longitude - longitude;
+        return Math.atan2(latitudeDifference, longitudeDifference);
```

Comment:
Is the comment still valid? atan2 smells like haversine, but comment says euclidean distance

Reply:
This entire getAngle() should not be there; I will polish the code before undrafting the PR.